app/design-system/design-system-hedvig/src/main/kotlin/com/hedvig/android/design/system/hedvig/ProgressBar.kt

```diff
@@ -0,0 +1,185 @@
+import com.hedvig.android.design.system.hedvig.ProgressDefaults.stroke
+import com.hedvig.android.design.system.hedvig.ProgressDefaults.strokeCap
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.FillPrimary
+import com.hedvig.android.design.system.hedvig.tokens.ColorSchemeKeyTokens.SurfaceSecondary
+import kotlin.math.abs
+
+@Composable
+fun HedvigLinearProgressBar(modifier: Modifier = Modifier) {
+  val infiniteTransition = rememberInfiniteTransition(label = "rememberInfiniteTransition")
+  val repeatable = infiniteRepeatable<Float>(
+    tween(progressAnimationDuration),
+    RepeatMode.Restart,
+  )
+  val animatedProgress = infiniteTransition.animateFloat(
+    0f,
+    1f,
+    repeatable,
+    label = "animatedProgress",
+  )
+
+  LinearProgressIndicator(
+    progress = { animatedProgress.value },
+    modifier = modifier.fillMaxWidth(),
+    color = progressColors.indicatorColor,
+    trackColor = progressColors.trackColor,
+    strokeCap = strokeCap,
+  )
+}
+
+@Composable
+private fun LinearProgressIndicator(
+  progress: () -> Float,
+  modifier: Modifier = Modifier,
+  color: Color,
+  trackColor: Color,
+  strokeCap: StrokeCap,
+) {
+  val coercedProgress = { progress().coerceIn(0f, 1f) }
+  Canvas(
+    modifier
+      .semantics(mergeDescendants = true) {
+        progressBarRangeInfo = ProgressBarRangeInfo(coercedProgress(), 0f..1f)
+      }
+      .height(stroke),
+  ) {
+    val strokeWidth = size.height
+    drawLinearIndicatorTrack(trackColor, strokeWidth, strokeCap)
+    drawLinearIndicator(0f, coercedProgress(), color, strokeWidth, strokeCap)
+  }
+}
+
+private fun DrawScope.drawLinearIndicator(
+  startFraction: Float,
+  endFraction: Float,
+  color: Color,
+  strokeWidth: Float,
+  strokeCap: StrokeCap,
+) {
+  val width = size.width
+  val height = size.height
+  val yOffset = height / 2
+
+  val isLtr = layoutDirection == LayoutDirection.Ltr
+  val barStart = (if (isLtr) startFraction else 1f - endFraction) * width
+  val barEnd = (if (isLtr) endFraction else 1f - startFraction) * width
+
+  if (strokeCap == StrokeCap.Butt || height > width) {
+    drawLine(color, Offset(barStart, yOffset), Offset(barEnd, yOffset), strokeWidth)
+  } else {
+    val strokeCapOffset = strokeWidth / 2
+    val coerceRange = strokeCapOffset..(width - strokeCapOffset)
+    val adjustedBarStart = barStart.coerceIn(coerceRange)
+    val adjustedBarEnd = barEnd.coerceIn(coerceRange)
+
+    if (abs(endFraction - startFraction) > 0) {
+      drawLine(
+        color,
+        Offset(adjustedBarStart, yOffset),
+        Offset(adjustedBarEnd, yOffset),
+        strokeWidth,
+        strokeCap,
+      )
+    }
+  }
+}
+
+private fun DrawScope.drawLinearIndicatorTrack(color: Color, strokeWidth: Float, strokeCap: StrokeCap) =
+  drawLinearIndicator(0f, 1f, color, strokeWidth, strokeCap)
+
+@Composable
+fun HedvigCircularProgressBar(modifier: Modifier = Modifier) {
```

Comment:
Yeah I see what you mean, it is close, but one thing that these progress bars typically do is that they are not a repeatable thing which does 1 loop on repeat, but their "starting point" moves along the circle a bit as the bar continues moving.
I think we need to either copy-paste the entire way they animate this from inside m3, or just use it as-is if it allows us to do the necessary adjustments to it, otherwise it does not "feel" right in a way.